### ISSUE TITLE
No loader able to handle the import of a .webmanifest file

### ISSUE THREAD

**Issue by acairns** [remix-run/remix#2954]
### What version of Remix are you using?

1.2.3

### Steps to Reproduce

Create a `manifest.webmanifest` file and import it.

Example contents:
```
{
  "icons": [
    { "src": "/icon-192.png", "type": "image/png", "sizes": "192x192" },
    { "src": "/icon-512.png", "type": "image/png", "sizes": "512x512" }
  ]
}
```

Then import it:
```
import manifest from "~/assets/manifest.webmanifest";
```

### Expected Behavior

It will import the file with the url, like mentioned in the docs:
https://remix.run/docs/en/v1/api/conventions#asset-url-imports

### Actual Behavior

Causes the following error:
```
✘ [ERROR] No loader is configured for ".webmanifest" files: app/assets/manifest.webmanifest

    app/root.tsx:4:21:
      4 │ import manifest from "~/assets/manifest.webmanifest";
        ╵                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Build failed with 1 error:
app/root.tsx:4:21: ERROR: No loader is configured for ".webmanifest" files: app/assets/manifest.webmanifest
```

**Comment by mcansh** [comment]
is there a reason you need to import this file instead of linking to it via the public folder, `<link rel="manifest" href="/manifest.webmanifest" />`? to my knowledge you shouldn't be caching this file for any extended amount of time

**Comment by mcansh** [comment]
on second thought this seems absolutely okay to import if you want to use any of the values - it's just json after all :)

I'm adding a test to the PR above and it should be released soon

**Comment by bewildergeist** [comment]
@mcansh Wait up, you're implementing support for loading the file as JSON in #3081, but what @acairns is asking for is "It will import the file with the url", i.e. as a fingerprinted asset URL.

I have the same use case — I would like to import and fingerprint the webmanifest, so I'm sure to cachebust it if anything in it changes. If I add it to the `/public` folder it will be served with a long-term `Cache-Control: public, max-age=31536000, immutable` header (at least when running the Remix App Server). This is exactly the opposite of your point of "you shouldn't be caching this file for any extended amount of time" 🙈

So being able to import a `*.webmanifest` file as a static asset would be great. Importing it as a JSON object, on the contrary, is not a use case I've had.

**Comment by mcansh** [comment]
@bewildergeist ah, i didn't see an explicit usage example and should've asked for clarification, I can see both options being useful, importing it as json to include as a route's links for example.

on the other-hand, adding it to the `/public` folder should only be caching for an hour if it's outside of the `/public/build`

**Comment by bewildergeist** [comment]
@mcansh:

> importing it as json to include as a route's links for example.

You mean including it in an exported `links` function? That's exactly the use case where you'd need to import it as a fingerprinted file URL though, right?

> adding it to the `/public` folder should only be caching for an hour if it's outside of the `/public/build`

Aw yeah, you're right! I hadn't noticed that change. I just upgraded from `1.3.1` to `1.4.3` so that must've been fixed somewhere in the interim — nice! 😄

Then you're right, importing it as a asset URL is perhaps less important. But I would still venture that most users would expect a `*.webmanifest` file to be `import`'ed as an asset URL rather than a JSON object 🤔

**Comment by mcansh** [comment]
@bewildergeist 
> You mean including it in an exported `links` function? That's exactly the use case where you'd need to import it as a fingerprinted file URL though, right?

i'm more so referring to setting the icons in the links function, for example: 
```ts
manifest.icons.map(icon => {
  return {
    rel: "alternate icon",
    type: "image/png",
    sizes: icon.sizes,
    href: icon.arc
  }
 });
```

**Comment by MichaelDeBoey** [comment]
The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/Manifest#deploying_a_manifest) are showing the file to be used in a `<link>` tag, so I think most people who are using this file will use it like that.

But I also like this tbh.
> ```ts
> manifest.icons.map(icon => {
>   return {
>     rel: "alternate icon",
>     type: "image/png",
>     sizes: icon.sizes,
>     href: icon.arc
>   }
>  });
> ```

@mcansh I guess it's not possible to support both use-cases?

**Comment by mcansh** [comment]
> @mcansh I guess it's not possible to support both use-cases?

i don't think so @MichaelDeBoey 😕 - at the time same time though, if we did import it as json, would you then import the file from the public directory, or maintain 2 copies 🤪

**Comment by MichaelDeBoey** [comment]
For me personally it would seem weird to get it from the `public` directory.

On the other hand it would also be weird to have 2 copies 😅

I never used this type of file myself, so I'm not a good person to ask these questions to I think 😛

**Comment by bewildergeist** [comment]
@MichaelDeBoey @mcansh: As it says in [the MDN docs](https://developer.mozilla.org/en-US/docs/Web/Manifest#deploying_a_manifest):

> Browsers generally support manifests with other appropriate extensions like `.json` (`Content-Type: application/json`)

So it feels reasonable to say:

1. If you want to just import an asset URL, use an `app.webmanifest` file
2. If you _also_ want to import and parse the file to e.g. extract icons, then use a `manifest.json` file

...which the #3081 PR (with Logan's recent update) would support.

(With the caveat that I haven't empirically tested whether "browsers generally support" _actually_ means that all current browsers support JSON manifests, but I assume so).

And the 3rd option would be to create e.g. a `app[.]webmanifest.js` route file that returns a `json()` response, which allows for custom `Cache-Control` headers and the ability to share e.g. icon references across the app. Like @jacob-ebey [does here](https://github.com/jacob-ebey/remix-pwa/blob/main/app/routes/resources/manifest%5B.%5Djson.ts).

**Comment by sergiodxa** [comment]
Since the .webmanifest file is a static file, any URL inside it would also be static, so I think importing it should give your a URL instead of the content, if you want to preload the icons you can know the URLs because they are static anyway.

**Comment by MichaelDeBoey** [comment]
Closed by #3081
